### ISSUE TITLE
Build errors on recent nightlies when compiling rustc-std-workspace-core on UEFI targets

### ISSUE THREAD

**Issue by IsaacWoods** [rust-lang/rust#71880]
A build error was reported in one of my projects (IsaacWoods/pebble#54) with recent nightlies (`rustc 1.45.0-nightly (fa51f810e 2020-04-29)` is the last known good nightly) using the target `x86_64-unknown-uefi`. The errors were:
```
   Compiling rustc-std-workspace-core v1.99.0 (/home/mtnygard/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/tools/rustc-std-workspace-core)
error: <inline asm>:2:33: error: unknown flag
            .section .llvmbc,"e"
                                ^

error: <inline asm>:3:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^

error: aborting due to 2 previous errors

error: could not compile `rustc-std-workspace-core`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: <inline asm>:2:33: error: unknown flag
            .section .llvmbc,"e"
                                ^

error: <inline asm>:3:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^

error: <inline asm>:2:33: error: unknown flag
            .section .llvmbc,"e"
                                ^

error: <inline asm>:3:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^

error: <inline asm>:55:33: error: unknown flag
            .section .llvmbc,"e"
                                ^

error: <inline asm>:56:34: error: unknown flag
            .section .llvmcmd,"e"
                                 ^
```

It looks like this issue was introduced by #71528. I will submit a PR that I believe fixes this on UEFI targets.

**Comment by JohnTitor** [comment]
It seems the issue has been fixed by #71881, closing.
